From ee52128fbb8bc953c43b8c53763443c715fa64c8 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Fri, 15 Apr 2022 07:29:49 +0200
Subject: [PATCH 16/26] bluetooth: support AT+CGMM

Return the modem model if ModemManager is available when the HF
sends an out-of-spec AT+CGMM command.
---
 src/modules/bluetooth/backend-native.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/modules/bluetooth/backend-native.c b/src/modules/bluetooth/backend-native.c
index 2fa769d07..2c39a090c 100644
--- a/src/modules/bluetooth/backend-native.c
+++ b/src/modules/bluetooth/backend-native.c
@@ -790,6 +790,13 @@ static bool hfp_rfcomm_handle(int fd, pa_bluetooth_transport *t, const char *buf
         }
         rfcomm_write_error(discovery->native_backend, fd, CMEE_NO_CONNECTION_TO_PHONE);
         return false;
+    } else if (strstr(buf, "AT+CGMM") && pa_modemmanager_has_modem(discovery->native_backend->modemmanager)) {
+        if (discovery->native_backend->modemmanager) {
+            rfcomm_write_response(fd, "%s", pa_modemmanager_get_modem_model(discovery->native_backend->modemmanager));
+            return true;
+        }
+        rfcomm_write_error(discovery->native_backend, fd, CMEE_NO_CONNECTION_TO_PHONE);
+        return false;
     }
 
     /* first-time initialize selected codec to CVSD */
-- 
2.35.1

